Client/Frontend/Home/HomePanelViewController.swift

```diff
@@ -194,4 +207,40 @@ class HomePanelViewController: UIViewController, UITextFieldDelegate, HomePanelD
     func homePanelDidRequestToSignIn(homePanel: HomePanel) {
         delegate?.homePanelViewControllerDidRequestToSignIn(self)
     }
+
+    func homePanelWillEnterEditingMode(homePanel: HomePanel) {
+        editingPanel = homePanel
+        toggleEditingMode(true)
+    }
+
+    private func toggleEditingMode(editing: Bool) {
+        let translateUp = CGAffineTransformMakeTranslation(0, -AppConstants.ToolbarHeight)
+        let translateDown = CGAffineTransformMakeTranslation(0, AppConstants.ToolbarHeight)
+
+        if editing {
+            finishEditingButton = {
+                let button = UIButton.buttonWithType(UIButtonType.System) as! UIButton
+                button.setTitle(NSLocalizedString("Done", comment: "Done editing button"), forState: UIControlState.Normal)
+                button.addTarget(self, action: "SELtappedEndEditing:", forControlEvents: UIControlEvents.TouchUpInside)
+                button.transform = translateDown
+                self.buttonContainerView.addSubview(button)
+                button.snp_makeConstraints { make in
+                    make.left.equalTo(self.buttonContainerView).offset(HomePanelViewControllerUX.EditDoneButtonLeftPadding)
+                    make.centerY.equalTo(self.buttonContainerView)
+                }
+                self.buttonContainerView.layoutIfNeeded()
+                return button
+            }()
```

Comment:
Any reason you're using this self-executing block syntax to set `finishEditingButton` rather than just doing this sequentially? That is,

```
if editing {
    ...
    self.buttonContainerView.layoutIfNeeded()
    finishEditingButton = button
}
```